src/user/user.service.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { UserService } from "./user.service";
+import { UserAuthService } from "../user-auth/user-auth.service";
+import { CreateUserDto } from "./dto/create-user.dto";
+import { UserAuth } from "../user-auth/interfaces/user-auth.interface";
+import { TestDatabaseModule } from "../test-database/test-database.module";
+import { MongooseModule } from "@nestjs/mongoose";
+import { UserSchema } from "./schemas/user.schema";
+
+class MockUserAuthService {
+  async create() {
+    /* Should be overriden using spyOn */
+  }
+  async findByEmail() {
+    /* Should be overriden using spyOn */
+  }
+  async findById() {
+    /* Should be overriden using spyOn */
+  }
+}
+
+describe("UserService", () => {
+  let service: UserService;
+  let userAuthService: UserAuthService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestDatabaseModule,
+        MongooseModule.forFeature([{ name: "User", schema: UserSchema }])
+      ],
+      providers: [
+        UserService,
+        { provide: UserAuthService, useValue: new MockUserAuthService() }
+      ]
+    }).compile();
+
+    service = module.get<UserService>(UserService);
+    userAuthService = module.get<UserAuthService>(UserAuthService);
+  });
+
+  it("should be defined", () => {
+    expect(service).toBeDefined();
+  });
+
+  it("should be able to create a user", async () => {
+    const createPayload = {
+      userAuthId: "random",
+      name: "George Blah"
+    } as CreateUserDto;
+
+    jest.spyOn(userAuthService, "findById").mockImplementation(async () => {
+      return ({
+        email: "george@example.com",
+        id: "random",
+        isVerified: true,
+        authType: "EmailAndPassword",
+        passwordHash: "random"
+      } as unknown) as UserAuth;
```

Comment:
Does this need to be casted to `unknown` before `UserAuth`? I tried casting it directly to `UserAuth` and the tests passed for me.